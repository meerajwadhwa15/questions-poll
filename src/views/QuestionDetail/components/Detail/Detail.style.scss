.choices {
    margin: 10px 0;
    color: $brand-black;
}