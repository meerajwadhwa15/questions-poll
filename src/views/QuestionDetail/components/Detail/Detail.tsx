import React, { FunctionComponent } from "react";
import { Question } from "@/types/questions";
import Choice from './../Choice';
import { dateFormater } from "@/utils";
import { Title } from "@/components";

export type Props = {
  item: Question;
};

const Detail: FunctionComponent<Props> = ({ item: { question, published_at, choices } }) => (
  <div className="question-detail">
    <Title>Question: <strong>{question}</strong></Title>
    <p>Published Date: <strong>{dateFormater(published_at)}</strong></p>
    <div className="choices">
      {choices.map(choice => (<Choice key={choice.url} choice={choice} />))}
    </div>
  </div>
);

export default Detail;
