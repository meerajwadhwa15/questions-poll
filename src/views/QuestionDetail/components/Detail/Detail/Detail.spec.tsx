import React from "react";
import Detail from "../Detail";
import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

const question = {
    question: "Favourite programming language?",
    published_at: "2020-03-12T12:42:40.310135+00:00",
    url: "/questions/1",
    choices: [
        {
            choice: "Swift",
            votes: 39,
            url: "/questions/1/choices/1"
        }
    ]
};

const props = {
    item: question,
};

test("Matches the snapshot with Questions Detail", () => {
    const component = create(
        <Detail {...props} />
    );
    expect(component.toJSON()).toMatchSnapshot();
});
