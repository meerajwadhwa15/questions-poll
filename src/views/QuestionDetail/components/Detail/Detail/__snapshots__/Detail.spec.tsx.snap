// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot with Questions Detail 1`] = `
<div
  className="question-detail"
>
  <h1
    className="page-title"
  >
    Question: 
    <strong>
      Favourite programming language?
    </strong>
  </h1>
  <p>
    Published Date: 
    <strong>
      12/Jun/2020
    </strong>
  </p>
  <div
    className="choices"
  >
    <div
      className="choice"
    >
      <div>
        Choice: 
        <strong>
          Swift
        </strong>
      </div>
      <div>
        Votes: 
        <strong>
          39
        </strong>
      </div>
    </div>
  </div>
</div>
`;
